File: ios/SsdidWallet/Domain/Did/SsdidRegistryResolver.swift
import Foundation

final class SsdidRegistryResolver: DidResolver {
    private let registryApi: RegistryApi

    init(registryApi: RegistryApi) {
        self.registryApi = registryApi
    }

    func resolve(did: String) async throws -> DidDocument {
        return try await registryApi.resolveDid(did: did)
    }
}

File: android/app/src/main/java/my/ssdid/wallet/domain/did/SsdidRegistryResolver.kt
package my.ssdid.wallet.domain.did

import my.ssdid.sdk.domain.did.DidResolver
import my.ssdid.sdk.domain.model.DidDocument
import my.ssdid.wallet.domain.transport.RegistryApi

class SsdidRegistryResolver(private val registryApi: RegistryApi) : DidResolver {
    override suspend fun resolve(did: String): Result<DidDocument> = runCatching {
        registryApi.resolveDid(did)
    }
}
